src/Symfony/Component/HttpKernel/HttpCache/Store.php

```diff
@@ -317,14 +317,30 @@ private function getMetadata($key)
     /**
      * Purges data for the given URL.
      *
+     * This method purges both the HTTP and the HTTPS version of the cache entry.
+     *
      * @param string $url A URL
      *
-     * @return bool true if the URL exists and has been purged, false otherwise
+     * @return bool true if the URL exists with either HTTP or HTTPS scheme and has been purged, false otherwise
      */
     public function purge($url)
     {
-        $key = $this->getCacheKey(Request::create($url));
+        $http = preg_replace('#^https#', 'http', $url);
+        $https = preg_replace('#^http#', 'https', $url);
 
+        return $this->doPurge($http) || $this->doPurge($https);
```

Comment:
my thinking was that when either http or https was purged we report that something was purged. the chances that *both* http and https were in cache are small.

i think the most sane use case for this is when doing cache invalidation, e.g. with FOSHttpCache: you need to send a request to the specific host by ip, rather than the domain name, to target each server instance. but if you do that over https, you get certificate mismatches and thus want to use http.

apart from such special cases, your webserver should probably redirect requests from http to https rather than allow both.